scripts/grt.py

```diff
@@ -25,16 +33,61 @@
 default_config = os.path.abspath(os.path.join(os.path.dirname(__file__), 'grt.yml'))
 
 
-def _init(config):
-    if config.startswith('/'):
-        config = os.path.abspath(config)
+def resolve_location(config):
+    """
+    resolve_location takes in a dict with autodetect (bool) and hardcoded
+    latitude and longitude values (floats). The function calls a number of
+    external websites in order to resolve the host's IP address, and their
+    geographic location.
+    """
+    if config['autodetect']:
+        # Get public IP
+        log.info("Locating server")
+        try:
+            ip_address = urllib2.urlopen('https://icanhazip.com').read()
+        except (urllib2.HTTPError, urllib2.URLError) as err:
+            log.error("Could not contact IP detection service. %s", err)
+            return None
+
+        geolocation_api = 'http://ip-api.com/json/{0}'.format(ip_address)
+
+        try:
+            response = urllib2.urlopen(geolocation_api).read()
+        except (urllib2.HTTPError, urllib2.URLError) as err:
+            log.error("Could not contact location detection service. %s", err)
+            return None
+
+        # Construct or get the Location
+        json_geoloc = json.loads(response)
+        log.info("Server Located (%s, %s)", json_geoloc['lat'], json_geoloc['lon'])
+        return {
+            'lat': json_geoloc['lat'],
+            'lon': json_geoloc['lon'],
+        }
```

Comment:
yep, I remembered your PR to coudlaunch, saved me some time since you already had examples of using the service :)